### PR TITLE
2.16.1 Release: Fixes Changelog with relevant information

### PR DESCRIPTION
**Pull request by sourabh1007**
1. updated release version in release notes.
2. updated release version Release history table.
3. Closed a github issue mentioned in "Known Issues" table as it is fixed in latest version. But still keeping it here as this issue still exist in "minimum recomended version" (i.e. 2.16.0)

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -6,6 +6,13 @@ The **minimum recommended version is [2.16.0](#2.16.0)**.
 Because version 3 of the Azure Cosmos DB .NET SDK includes updated features and improved performance, we’ll retire version 2.x of this SDK on 31 August 2024.  As a result, you’ll need to update your SDK to version 3 by that date. We recommend following the [instructions](https://docs.microsoft.com/azure/cosmos-db/sql/migrate-dotnet-v3?tabs=dotnet-v3) to migrate to Azure Cosmos DB .NET SDK version 3.
 
 ## Release notes
+### <a name="2.16.1"></a> 2.16.1
+* Improved availability for Direct +TCP mode by setting EnableTcpConnectionEndpointRediscovery to on by default.
```
Comment (ealsur) [inline changelog.md:10]:
```suggestion
* Improved availability for Direct + TCP mode by setting EnableTcpConnectionEndpointRediscovery to `true` by default.
```

File: changelog.md [hunk 2]
```diff
@@ -6,6 +6,13 @@ The **minimum recommended version is [2.16.0](#2.16.0)**.
...
+* Improved retryWith(449) retry policy by having it retry sooner when multiple 449s are hit for the same request.
```
Comment (ealsur) [inline changelog.md:11]:
```suggestion
* Improved RetryWith(HTTP 449) retry mechanics by having it retry sooner when multiple 449s are hit for the same request.
```

File: changelog.md (hunk 1 above)
Comment (j82w) [inline changelog.md:10]:
```suggestion
* Improved availability for Direct + TCP mode by setting EnableTcpConnectionEndpointRediscovery to on by default.
```

File: changelog.md [hunk 3]
```diff
@@ -6,6 +6,13 @@ The **minimum recommended version is [2.16.0](#2.16.0)**.
...
+* Fixed queries going to gateway when SDK is configured with Direct + TCP. Only impacts queries without aggregates or no MaxDegreeOfParallelism is set with the ServiceInterop.dll not being available.
```
Comment (ealsur) [inline changelog.md:12]: Please update the Known Issues section, the table contains one issue for this, update the Mitigation column with the update to 2.16.1

**Pull request merged**